src/plays/star-rating/StarRating.js

```diff
@@ -0,0 +1,82 @@
+import PlayHeader from "common/playlists/PlayHeader";
+import { useState } from "react";
+import { FaStar } from "react-icons/fa";
+import "./styles.css";
+
+// WARNING: Do not change the entry componenet name
+function StarRating(props) {
+  // Your Code Start below.
+
+  const [rating, setRating] = useState(null);
+  const [hoverFill, setHoverFill] = useState(null);
+
+  const getReviewLabel = (rating) => {
+    switch (rating) {
+      case 1:
+        return `Very bad ${String.fromCodePoint("0x1F922")}`;
+      case 2:
+        return `Bad ${String.fromCodePoint("0x1F97A")}`;
+      case 3:
+        return `Okay ${String.fromCodePoint("0x1F60C")}`;
+      case 4:
+        return `Good ${String.fromCodePoint("0x1F60A")}`;
+      case 5:
+        return `Excellent ${String.fromCodePoint("0x1F929")}`;
+
+      default:
+        return "";
+    }
+  };
+
+  return (
+    <>
+      <div className="play-details">
+        <PlayHeader play={props} />
+        <div className="play-details-body">
+          {/* Your Code Starts Here */}
+
+          <div className="star-wrapper">
+            <h2 className="review-text">
+              {rating
+                ? getReviewLabel(rating)
```

Comment:
I noticed that, when i click on any star and after that when I hover on another one, the label doesn't get changed. It should be updated with hover and when mouse exists the section the label should go back to the last selection
![image](https://user-images.githubusercontent.com/7145967/187090493-a04f9d5d-b5a7-4ad0-93a6-97a6c05f5ede.png)

Reply:
Can you suggest how I go about this

Reply:
Something like

You woulld need another stat variable like `isHover`, this needs to be set from `onMouseEnter` and unset from `onMouseLeave`

and our styling will be 
```javascript

<FaStar
                      className="FaStar"
                      size={80}
                      style={{
                        color: (isHover && ratingValue <= hoverFill ) ||  (ratingValue  <= rating)? 
                          ratingValue <= (hoverFill || rating)
                            ? "#ffe101"
                            : "#ccc",
                      }}
                      onChange={() => setRating(ratingValue)}
                      value={ratingValue}
                    />
```

**Note**: Untested code. Its just to give the logic that would work

Reply:
Okay, thank you